### PR TITLE
[ICD] Add onTransitionToActiveMode Event to the ICDM cluster

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Adds the onTransitionToActiveMode event to the ICDM cluster
Implements https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8733

Event is enable for the Lit-ICD-app

#### Tests
No new tests. PR only has generation

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml:1]: Approving, but should probably wait for the spec PR to actually merge.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml:1]: Spec PR is only missing one review when you have a minute

**Review by yunhanw-google — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml:1]: Spec PR was merged. Resolving comment

Comment (yunhanw-google) [reply src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml:1]: merged spec pr.

**Pull request merged**